### ISSUE TITLE
Support custom props

### ISSUE THREAD

**Issue by Ky6uk** [Igorbek/typescript-plugin-styled-components#13]
The plugin doesn't compile code with type arguments for tagged template strings.

So the code below

```ts
export const Component = styled.div<{isDisabled?: boolean}>`
  color: ${props => props.isDisabled ? 'blue' : 'red' };
`;
```

will throw something like

```
Module parse failed: Unexpected token (18:14)
You may need an appropriate loader to handle this file type.
| `;
| export const Component = styled.div.withConfig({ displayName: "Component" })<{
>     isDisabled?: boolean;
| }> `
```

**Comment by Igorbek** [comment]
Thank you for reporting, I'll check it shortly.
Also, can you please verify that you use the single TS node module and plugin doesn't have its own local one.

**Comment by Igorbek** [comment]
Do you use Babel as a transpiler? Does the error come from it?

**Comment by Ky6uk** [comment]
@Igorbek I'm not using babel. Maybe `awesome-typescript-loader` does, but I'm not sure.

[There is](Ky6uk/tpsc-tagged-templates-bug) a minimal repo for reproducing the bug.

**Comment by Igorbek** [comment]
Thank you for this repro repo. I'm looking into this. It could be something to do with how atl handles results in the new version. Investigation is in progress.

**Comment by Igorbek** [comment]
A small update. I still did not isolate the issue. It has something to do with transpilation after the transformations for targets higher than ES5.
You can notice that bug disappears in case you change target to ES5 or lower.
Also, as a workaround, you can define interface and the bug disappears as well:
```diff
+interface Props { isDisabled?: boolean }
-export const Component = styled.div<{isDisabled?: boolean}>`
+export const Component = styled.div<Props>`
  color: ${props => props.isDisabled ? 'blue' : 'red' };
`;
```

**Comment by Inoir** [comment]
Dont wanted to open new nearly same issue, but im using this plugin now on TS 3.0.1! All is working out of the box, no errors, only warnings about incorrect peerDependencies.

typescript: 3.0.1
styled-components: 2.4.0

> warning "workspace-aggregator-5c177420-6ce0-46e8-aa52-2f6df0306aa0 > project > typescript-plugin-styled-components@0.0.6" has incorrect peer dependency "typescript@^2.5.2".

> warning "workspace-aggregator-5c177420-6ce0-46e8-aa52-2f6df0306aa0 > project > typescript-plugin-styled-components > ts-is-kind@1.0.0"has incorrect peer dependency "typescript@^2.4.2".

If youve a look also into [ts-is-kind](https://github.com/mohsen1/ts-is-kind), the lib says its deprecated, cause TS >v2.6 supports this. So maybe can also get rid of this dependency?!

**Comment by Igorbek** [comment]
Thank you for pointing this out @Inoir . I'll update the library in the next couple days.

**Comment by Igorbek** [comment]
@Inoir [a new release](https://github.com/Igorbek/typescript-plugin-styled-components/releases/tag/v0.0.7) with the fixes has been published.

**Comment by Igorbek** [comment]
Oh, I'm sorry I closed it. I'll have another look into this

**Comment by Igorbek** [comment]
It seems the issue in `awesome-typescript-loader@5`. When I downgraded it to 4.x and 3.x your repro code worked fine. I'll examine their code what was changed in v5.

**Comment by Igorbek** [comment]
Correction: it does not work actually.

**Comment by Igorbek** [comment]
It turned out to be a TS bug: https://github.com/Microsoft/TypeScript/issues/27854

**Comment by gtrufitt** [comment]
Hey @Igorbek, for clarity, do you think there is a workaround for this for the time-being without using the dev branch of TypeScript? Thanks!

**Comment by Ky6uk** [comment]
Just upgraded [repro repo](https://github.com/Ky6uk/tpsc-tagged-templates-bug). I'm not sure TS fix currently presented in `v3.1.6` but the bug still exists.

<details>

## System:
 - OS: macOS 10.14.1
 - CPU: (4) x64 Intel(R) Core(TM) i5-4250U CPU @ 1.30GHz
 - Memory: 130.05 MB / 8.00 GB
 - Shell: 4.4.23 - /usr/local/bin/bash
## Binaries:
 - Node: 11.1.0 - /usr/local/bin/node
 - Yarn: 1.12.3 - /usr/local/bin/yarn
 - npm: 6.4.1 - /usr/local/bin/npm
 - Watchman: 4.9.0 - /usr/local/bin/watchman
## npmPackages:
 - @types/node: 10.12.5 => 10.12.5
 - @types/react: 16.7.3 => 16.7.3
 - @types/styled-components: 3.0.2 => 3.0.2
 - awesome-typescript-loader: 5.2.1 => 5.2.1
 - react: 16.6.1 => 16.6.1
 - styled-components: 3.4.10 => 3.4.10
 - typescript: 3.1.6 => 3.1.6
 - typescript-plugin-styled-components: 1.0.0 => 1.0.0
 - webpack: 4.25.1 => 4.25.1
 - webpack-command: 0.4.2 => 0.4.2

</details>

**Comment by Igorbek** [comment]
Unfortunately [the fix](https://github.com/Microsoft/TypeScript/pull/27964) will go to TypeScript 3.2 only.

**Comment by Igorbek** [comment]
BTW you can try it out with `typescript@next` or `typescript@dev`

**Comment by Ky6uk** [comment]
Yep, just tested it on TS `3.2.0-rc` and it's working fine.

**Comment by Igorbek** [comment]
Fixed with [TypeScript 3.2](https://github.com/Microsoft/TypeScript/releases/tag/v3.2.1).
